### PR TITLE
community/pdns: Upgrade to 4.1.1

### PR DESCRIPTION
**Pull request by tcely**
Updated license.
Add additional backends.
Fixed default configuration directory.
Add unit tests and checks.
Added PDNS_TEST_NO_IPV6=1 to fix Travis CI test failure.
Add tools subpackage

### CODE REVIEW

**Comment by tcely** [comment]
The travis-ci build is broken because of a bad file.

```d74bcef3e735:/var/cache/apk$ for _apk in mariadb-dev-10.1.28-r1*.apk; do echo "$_apk"; sha512sum "$_apk" | awk '{print $1}'; >/dev/null tar -tf "$_apk"; echo; done; unset -v _apk
mariadb-dev-10.1.28-r1.dl-2.apk
a84d3095cbc8a1a2b56884df8b32e0861b85a240749e5c8d4683767a5ef6ca6e8bfa518aea69da12ca972dc4d38464627e69ae19e066cd262763731cdd394297

mariadb-dev-10.1.28-r1.dl-3.apk
a84d3095cbc8a1a2b56884df8b32e0861b85a240749e5c8d4683767a5ef6ca6e8bfa518aea69da12ca972dc4d38464627e69ae19e066cd262763731cdd394297

mariadb-dev-10.1.28-r1.dl-cdn.apk
a84d3095cbc8a1a2b56884df8b32e0861b85a240749e5c8d4683767a5ef6ca6e8bfa518aea69da12ca972dc4d38464627e69ae19e066cd262763731cdd394297

mariadb-dev-10.1.28-r1.nl.apk
47afd32c54dbf4ffea74f534c80204ca3f2932905298c3a9281b540737c89b341cc0bf88e16f171e9b41c404572f3c7120d1003e0e887a6cde9f97ab77d025ad
gzip: crc error
tar: Child returned status 1
tar: Error is not recoverable: exiting now

d74bcef3e735:/var/cache/apk$
```

File: community/pdns/APKBUILD [hunk 1]
```diff
@@ -39,48 +47,80 @@ build() {
...
 
 	make DESTDIR="$pkgdir" install
 
-	install -m755 -D "$srcdir"/$pkgname.initd \
-		"$pkgdir"/etc/init.d/$pkgname
-	rm "$pkgdir"/etc/$pkgname/$pkgname.conf-dist
-	install -m600 -D "$srcdir"/$pkgname.conf \
-		"$pkgdir"/etc/$pkgname/$pkgname.conf
-	chown pdns:pdns "$pkgdir"/etc/$pkgname/$pkgname.conf
+	install -m755 -D "${srcdir}/${pkgname}.initd" \
```
Comment (jirutka) [inline community/pdns/APKBUILD]: Why did you add unnecessary curly braces around variables? Please don’t do that, it just increases verbosity.

Comment (tcely) [reply community/pdns/APKBUILD]: I prefer adding the braces because it's more explicit and easier for us humans to interpret. Once you're in the habit of using quotes and braces all the time, your shell code is usually safer and less ambiguous.

```sh
printf -- 'With:\n\t%s\nWithout:\n\t%s\n\n' \
    "We're having ${fruit}s and ${nut}s after lunch." \
    "We're having $fruits and $nuts after lunch."
With:
        We're having s and s after lunch.
Without:
        We're having  and  after lunch.


fruit='bananna'
nut='pecan'

printf -- 'With:\n\t%s\nWithout:\n\t%s\n\n' \
    "We're having ${fruit}s and ${nut}s after lunch." \
    "We're having $fruits and $nuts after lunch."
With:
        We're having banannas and pecans after lunch.
Without:
        We're having  and  after lunch.


fruits='apples, oranges,'

printf -- 'With:\n\t%s\nWithout:\n\t%s\n\n' \
    "We're having ${fruit}s and ${nut}s after lunch." \
    "We're having $fruits and $nuts after lunch."
With:
        We're having banannas and pecans after lunch.
Without:
        We're having apples, oranges, and  after lunch.


nuts='peanuts'

printf -- 'With:\n\t%s\nWithout:\n\t%s\n\n' \
    "We're having ${fruit}s and ${nut}s after lunch." \
    "We're having $fruits and $nuts after lunch."
With:
        We're having banannas and pecans after lunch.
Without:
        We're having apples, oranges, and peanuts after lunch.
```

Comment (jirutka) [reply community/pdns/APKBUILD]: I disagree. It may improve readability and clarity in _some cases_, but that’s not a reason for adding them _everywhere_. We have syntax highlighters in editors to highlight variables for humans. Adding curly braces around _all_ variables does not make shell scripts safer or less ambiguous, unless you don’t know what is a valid variable name in shell (for example, that variable name can never contain a space). Also changing this in existing abuilds just creates unnecessary noise.

Please follow our current code-style and do not force your habits into abuilds. Otherwise I’d have to reject your pull requests. Thank you for understanding.
